### PR TITLE
[Refactor] Replace <ArtworkFilter> in /collect 

### PR DESCRIPTION
**Pull request by damassi**
This PR swaps in the new generic ArtworkFilter component into `/collect`, and rounds out the new `ArtworkFilter` API.

#### Todo: 
- [x] Tests 
- [x] Add toLowerCase on filter urls -- `works-on-Paper` errors out in MP
- [x] Fix time period filter; state doesn't update and padding is off (major_periods array)

#### Example

```tsx
<ArtworkFilter
  viewer={viewer}
  filters={{ foo, bar, baz }}
  sortOptions={[
    { value: "-decayed_merch", text: "Default" },
    { value: "-partner_updated_at", text: "Recently updated" },
    { value: "-published_at", text: "Recently added" },
    { value: "-year", text: "Artwork year (desc.)" },
    { value: "year", text: "Artwork year (asc.)" },
  ]}
  onChange={filters => {
    const url = buildUrlForCollectApp(filters)

    if (typeof window !== "undefined") {
      window.history.pushState({}, null, url)
    }
  }}
  onFilterClick={(key, value, filterState) => {
    trackEvent({
      action_type: Schema.ActionType.CommercialFilterParamsChanged,
      changed: { [key]: value },
      current: filterState,
    })
  }}
  onBrickClick={props => doStuff}
  ZeroState={SomeCustomZeroState} 
/>
```

filter2

### CODE REVIEW

File: src/Apps/Collect2/Routes/Collect/index.tsx [hunk 1]
```diff
@@ -1,153 +1,136 @@
...
-            Collect art and design online
+        <Box mt={3}>
+          <Serif size="8">
+            <h1>Collect art and design online</h1>
           </Serif>
-          <Separator mt={2} mb={[2, 2, 2, 4]} />
 
           {showCollectionHubs && (
             <>
+              <Separator mt={2} mb={[2, 2, 2, 4]} />
```
Comment (pepopowitz) [inline src/Apps/Collect2/Routes/Collect/index.tsx]: Are you sure we don't want to render this when the hubs aren't showing? I'd think we still want a separator between the `h1` and grid.

File: src/Apps/Collect2/Routes/Collect/index.tsx [hunk 2]
```diff
@@ -1,153 +1,136 @@
...
+        </Box>
 
-          <ArtworkGrid viewer={this.props.viewer} />
-        </FrameWithRecentlyViewed>
-      </AppContainer>
-    )
-  }
-}
+        <Box>
+          <ArtworkFilter
```
Comment (pepopowitz) [inline src/Apps/Collect2/Routes/Collect/index.tsx]: 💪

File: src/Artsy/Router/RouterLink.tsx [hunk 3]
```diff
@@ -19,15 +20,38 @@ export const RouterLink: React.FC<LinkProps> = (
 ) => {
   const isRouterContext = Boolean(context.router)
 
+  // Only pass found-router specific props across
```
Comment (pepopowitz) [inline src/Artsy/Router/RouterLink.tsx]: I think this comment is either incorrect, or could use more definition of what is a found-specific prop. Very minor, but if I come back here in the future, I think there's a mismatch between what I'd read in the comment and which props are being passed through.

File: src/Components/v2/ArtworkFilter/ArtworkFilterArtworkGrid2.tsx [hunk 4]
```diff
@@ -1,36 +1,31 @@
...
+import { useSystemContext } from "Artsy"
+import ArtworkGrid from "Components/ArtworkGrid"
 import { LoadingArea } from "../LoadingArea"
 import { PaginationFragmentContainer as Pagination } from "../Pagination"
-import { useFilterContext } from "./ArtworkFilterContext"
-import { ArtworkFilterZeroState } from "./ArtworkFilterZeroState"
-
-// TODO: Wire up
-// import { SortFilter } from "./ArtworkFilterArtworkGridSort"
+import { useArtworkFilterContext } from "./ArtworkFilterContext"
```
Comment (pepopowitz) [inline src/Components/v2/ArtworkFilter/ArtworkFilterArtworkGrid2.tsx]: Good name change 👍

File: src/Components/v2/ArtworkFilter/ArtworkFilterContext.tsx [hunk 5]
```diff
@@ -32,8 +30,23 @@ export interface Filters {
...
+
+  // Handlers
+  onArtworkBrickClick?: (artwork: any, props: any) => void
+  onFilterClick?: (
+    key: keyof ArtworkFilters,
+    value: string,
+    filterState: ArtworkFilters
+  ) => void
+
+  // Filter manipulationo
```
Comment (pepopowitz) [inline src/Components/v2/ArtworkFilter/ArtworkFilterContext.tsx]:
```suggestion
  // Filter manipulation
```

(spelling)

File: src/Artsy/Router/Utils/Route.tsx [hunk 6]
```diff
@@ -47,9 +47,33 @@ function createRender({
...
+          In attempting to avoid the use of <StaticContainer> in RenderStatus.tsx,
+          which freezes the component tree with `shouldComponentUpdate = false`,
+          we stored the previously-rendered component and props in a variable and
+          instead of returning undefined here, we returned <PrevComponent {...prevProps} />.
+
+          However, when the component is rendered by react, it errors out because
+          the data in prevProps has seemingly been garbage collected.
+
+          Relay has the ability to `retain` data in the store. We should investigate,
+          which wouold give us greater control over our component tree when top-level
```
Comment (mzikherman) [inline src/Artsy/Router/Utils/Route.tsx]: typo *would*

**Comment by mzikherman** [comment]
Took the collect page for a little walk in Force while linked, looks great!

One note for posterity:
  - `?major_periods[]=...` instead of `major_periods=...` (the array form is correct) when clicking the period radio button. It doesn't 'stick' and the URL is updated w/o the array form.

File: src/Apps/Collect2/Routes/Collect/index.tsx (hunk 1 above)
Comment (damassi) [reply src/Apps/Collect2/Routes/Collect/index.tsx]: Ah, nice  catch 👍

**Comment by artsyit** [comment]
:rocket: PR was released in v17.5.0 :rocket:

**Pull request merged**